lib/node.js

```diff
@@ -211,22 +210,38 @@ class Node {
       case 'role':
         delete this.kuzzle.repositories.role.roles[data.id];
         break;
-      case 'strategy:added':
-        try {
-          this.kuzzle.pluginsManager.registerStrategy(data.pluginName, data.name, data.strategy);
-        } catch (e) {
-          // log & discard
-          this.context.log('error', `Plugin ${data.pluginName} - tried to add the strategy "${data.name}": ${e.message}`);
-        }
-        break;
-      case 'strategy:removed':
-        try {
-          this.kuzzle.pluginsManager.unregisterStrategy(data.pluginName, data.name);
-        } catch (e) {
-          // log & discard
-          this.context.log('error', `Plugin ${data.pluginName} - tried to remove the strategy "${data.name}": ${e.message}`);
-        }
-        break;
+      case 'strategies':
+        return this.redis.hgetall('cluster:strategies')
+          .then(strategies => {
+            const currentStrategies = new Set(this.kuzzle.pluginsManager.listStrategies());
+
+            if (!strategies) {
+              strategies = {};
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "strategies". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1226)